massa-execution-worker/src/speculative_ledger.rs

```diff
@@ -138,14 +156,41 @@ impl SpeculativeLedger {
         // simulate crediting coins to destination address (if any)
         // note that to_addr can be the same as from_addr
         if let Some(to_addr) = to_addr {
-            let new_balance = changes
-                .get_balance_or_else(&to_addr, || self.get_balance(&to_addr))
-                .unwrap_or_default()
-                .checked_add(amount)
-                .ok_or_else(|| {
+            let old_balance = changes.get_balance_or_else(&to_addr, || self.get_balance(&to_addr));
+            if let Some(old_balance) = old_balance {
+                let new_balance = old_balance.checked_add(amount).ok_or_else(|| {
                     ExecutionError::RuntimeError("overflow in to_addr balance".into())
                 })?;
-            changes.set_balance(to_addr, new_balance);
+                changes.set_balance(to_addr, new_balance);
+            } else if from_addr.is_some() {
+                // Debit sender to create receiver address
+                let address_storage_cost = self
```

Comment:
using the LEDGER_ENTRY_BASE_SIZE mentionned above avoids having to multiply by the address size and adding ledger cost for balance